src/Symfony/Component/DependencyInjection/Tests/Compiler/ResolveDefinitionTemplatesPassTest.php

```diff
@@ -139,6 +139,25 @@ public function testProcessDoesNotCopyDecoratedService()
         $this->assertNull($def->getDecoratedService());
     }
 
+    public function testProcessDoesNotDropShared()
```

Comment:
Please also add a ``testProcessDoesNotCopyShared``, to mimic what we have for the scope